onmt/trainer.py

```diff
@@ -52,7 +52,7 @@ def build_trainer(opt, device_id, model, fields, optim, model_saver=None):
                            shard_size, norm_method,
                            grad_accum_count, n_gpu, gpu_rank,
                            gpu_verbose_level, report_manager,
-                           model_saver=model_saver)
+                           model_saver=model_saver if gpu_rank == 0 else None)
```

Comment:
This condition is to meant avoid saving on each running process. I'm not sure if this is checked elsewhere.

Reply:
It seems clearer to not pass the model saver object if a `Trainer` instance should not save any models. Do you think there is an issue with this code?

Reply:
that's perfectly correct, my bad (not sure how it's even possible but I understood you removed that code, when you precisely added it for the reason I mention...)